Core/include/Acts/Seeding/EstimateTrackParamsFromSeed.hpp

```diff
@@ -0,0 +1,253 @@
+  Vector3 bGlobal(sps[0]->x(), sps[0]->y(), sps[0]->z());
+  Vector3 mGlobal(sps[1]->x(), sps[1]->y(), sps[1]->z());
+  Vector3 tGlobal(sps[2]->x(), sps[2]->y(), sps[2]->z());
+  // Define a new coordinate frame with its origin at the bottom space point, z
+  // axis long the magnetic field direction and y axis perpendicular to vector
+  // from the bottom to middle space point. Hence, the projection of the middle
+  // space point on the tranverse plane will be located at the x axis of the new
+  // frame.
+  Vector3 mRelVec = mGlobal - bGlobal;
+  Vector3 newZAxis = bField.normalized();
+  Vector3 newYAxis = newZAxis.cross(mRelVec).normalized();
+  Vector3 newXAxis = newYAxis.cross(newZAxis);
+  RotationMatrix3 rotation;
+  rotation.col(0) = newXAxis;
+  rotation.col(1) = newYAxis;
+  rotation.col(2) = newZAxis;
+  // The center of the new frame is at the bottom space point
+  Translation3 trans(bGlobal);
+  // The transform which constructs the new frame
+  Transform3 transform(trans * rotation);
+
+  // The coordinate of the middle and top space point in the new frame
+  Vector3 local1 = transform.inverse() * mGlobal;
+  Vector3 local2 = transform.inverse() * tGlobal;
+
+  // Lambda to transform the coordinates to the (u, v) space
+  auto uvTransform = [](const Vector3& local) -> Vector2 {
+    Vector2 uv;
+    ActsScalar denominator = local.x() * local.x() + local.y() * local.y();
+    uv.x() = local.x() / denominator;
+    uv.y() = local.y() / denominator;
+    return uv;
+  };
+  // The uv1.y() should be zero
+  Vector2 uv1 = uvTransform(local1);
+  Vector2 uv2 = uvTransform(local2);
+
+  // A,B are slope and intercept of the straight line in the u,v plane
+  // connecting the three points.
+  ActsScalar A = uv2.y() / (uv2.x() - uv1.x());
+  ActsScalar B = uv2.y() - A * uv2.x();
+  // Curvature (with a sign) estimate
+  ActsScalar rho = -2.0 * B / std::hypot(1., A);
+  // The projection of the top space point on the transverse plane of the new
+  // frame
+  ActsScalar rn = local2.x() * local2.x() + local2.y() * local2.y();
+  // The (1/tanTheta) of momentum in the new frame, corrected for curvature
+  // effects (@note why 0.04?)
+  ActsScalar invTanTheta =
+      local2.z() * std::sqrt(1. / rn) / (1. + 0.04 * rho * rho * rn);
+  // The momentum direction in the new frame (the center of the circle has the
+  // coordinate (-1.*A/(2*B), 1./(2*B)))
+  Vector3 transDirection(1., A, std::hypot(1, A) * invTanTheta);
+  // Transform it back to the original frame
+  Vector3 direction = rotation * transDirection.normalized();
+
+  // The phi and theta
+  params[eBoundPhi] = VectorHelpers::phi(direction);
+  params[eBoundTheta] = VectorHelpers::theta(direction);
+
+  // Transform the bottom space point to the provided surface
+  auto lpResult = surface.globalToLocal(gctx, bGlobal, direction);
+  if (not lpResult.ok()) {
+    ACTS_ERROR("Global to local transformation did not succeed.");
+    return std::nullopt;
+  }
```

Comment:
Yes, the bottom space point is assumed to be on the provided surface here. I modified the document to make this more clear.
In principle, I think the surface could be any surface provided by the user. But since we don't have a propagator to extrapolate the track parameters to a provided surface, I just constrain it to be at the bottom space point.

Reply:
Ok. I added some suggestions to make this clear in the documentation.